### PR TITLE
fix-9992 - Android - CollectionView - VerticalItemSpacing and HorizontalItemSpacing was not set for the new added item

### LINKED ISSUE #9992: [Bug] [Android] [CollectionView]  Add new element to ItemSource bypasses VerticalItemSpacing
### Description
When using a CollectionView on Android (iPhone works correctly) with "GridItemsLayout VerticalItemSpacing" > 0 the vertical spacing is ignored if the observed ItemSource gets a new element (e.g. ObservableCollection<T>.Add)
Refreshing the list (clear + add) solves the issue.

### Steps to Reproduce
1. Have a CollectionView with a DataTemplate for its items and GridItemsLayout with VerticalItemSpacing > 0 
2. Add new Items to the observed collection.

### Expected Behavior
New items added to the ItemSource must keep the vertical spacing among one another
Screenshot_1584455887

### Basic Information

- Version with issue: at least since 4.4 (Sample project runs XF 4.5.0.396)
- Last known good version: N.A.
- IDE: VS for Mac 8.4.8 (build 2)
- Platform Target Frameworks: devices with API 23 -> 28 for sure.
- Xamarin.Android: Version: 10.1.3.7 (Visual Studio Professional)
- Android Support Library Version: Recently added AndroidX

### Screenshots
Screenshot_1584455880

### Reproduction Link
[CollectionViewBug2.zip](https://github.com/xamarin/Xamarin.Forms/files/4344000/CollectionViewBug2.zip)

### Workaround
Clear the observed ItemSource, add all items all over again.

### PR DESCRIPTION
**Pull request by BrayanKhosravian**
### Description of Change ###
fix-9992 - Android - CollectionView - VerticalItemSpacing and HorizontalItemSpacing was not set for the new added item

### Issues Resolved ### 

- fixes #9992

### API Changes ###
 None

### Platforms Affected ### 
- Android

### Behavioral/Visual Changes ###
Android - CollectionView - VerticalItemSpacing and HorizontalItemSpacing was not set for the new added item

### Broken
vertical_broken

horizontal_broken

### Fixed
vertical_fixed

horizontal_fixed

Not applicable

### Testing Procedure ###
Just add a new item to the `CollectionView` and check if the item spacing is applied

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.Platform.Android/CollectionView/SpacingItemDecoration.cs [hunk 1]
```diff
@@ -74,17 +74,32 @@ public override void GetItemOffsets(ARect outRect, AView view, RecyclerView pare
...
 				outRect.Left = spanIndex == 0 ? 0 : (int)_adjustedHorizontalSpacing;
+				
+				var vertical = (int)_adjustedVerticalSpacing;
+				if (vertical != 0)
+				{
+					outRect.Top = vertical;
+					outRect.Bottom = vertical;
+				}
 
 				if (parent.GetChildAdapterPosition(view) != parent.GetAdapter().ItemCount - 1)
```
Comment (BrayanKhosravian) [inline Xamarin.Forms.Platform.Android/CollectionView/SpacingItemDecoration.cs:87]: I don't understand why this exists. `if (parent.GetChildAdapterPosition(view) != parent.GetAdapter().ItemCount - 1)`

File: Xamarin.Forms.Platform.Android/CollectionView/SpacingItemDecoration.cs [hunk 2]
```diff
@@ -74,17 +74,32 @@ public override void GetItemOffsets(ARect outRect, AView view, RecyclerView pare
...
 
+				var horizontal = (int)_adjustedHorizontalSpacing;
+
+				if (horizontal != 0)
+				{
+					outRect.Left = horizontal;
+					outRect.Right = horizontal;
+				}
+
 				if (parent.GetChildAdapterPosition(view) != parent.GetAdapter().ItemCount - 1)
```
Comment (BrayanKhosravian) [inline Xamarin.Forms.Platform.Android/CollectionView/SpacingItemDecoration.cs:101]: Same for this one. "I don't understand why this exists." `if (parent.GetChildAdapterPosition(view) != parent.GetAdapter().ItemCount - 1)`

**Review by BrayanKhosravian — COMMENTED** [review]
How should this be tested?

**Comment by BrayanKhosravian** [comment]
I think it would have been better for the history to do a rebase.

**Review by jsuarezruiz — CHANGES_REQUESTED** [review]
Could you rebase to fix the current conflict?. Thanks!.

**Comment by BrayanKhosravian** [comment]
@jsuarezruiz Should I rebase onto 5.0.0 or another branch?

**Comment by jfversluis** [comment]
Hey @BrayanKhosravian thank you so much for your time and efforts on this. I am so sorry we kept this around for so long. I have checked and it seems this got fixed in the meantime with another PR already. Despite everything I hope you'll join us with .NET MAUI where we're improving all kinds of things, including the process for PRs (from the community).

Closing this one for now, let me know if you still thing something is wrong in this area. Thanks!

**Pull request closed without merging**